### PR TITLE
Asks for write permission in runtime on Android 6+

### PR DESCRIPTION
**Pull request by SidhNor**

### CODE REVIEW

File: Android/ChorusRFLaptimer/app/src/main/java/app/andrey_voroshkov/chorus_laptimer/MainActivity.java [hunk 1]
```diff
@@ -117,6 +122,8 @@ protected void onCreate(Bundle savedInstanceState) {
         AppState.getInstance().preferences = getPreferences(MODE_PRIVATE);
         AppPreferences.applyAll();
 
+        //Ensure permissions permissions before any disk IO
+        ensurePermissions();
```
Comment (voroshkov) [inline Android/ChorusRFLaptimer/app/src/main/java/app/andrey_voroshkov/chorus_laptimer/MainActivity.java]: should we do the same before creating the report?

Comment (SidhNor) [reply Android/ChorusRFLaptimer/app/src/main/java/app/andrey_voroshkov/chorus_laptimer/MainActivity.java]: Nope, as this is in main activity, it truly need to be requested once.

Comment (SidhNor) [reply Android/ChorusRFLaptimer/app/src/main/java/app/andrey_voroshkov/chorus_laptimer/MainActivity.java]: We could actually ask this permission only when writing the report as if the permission wasn't granted, there are no reports to remove

Comment (voroshkov) [reply Android/ChorusRFLaptimer/app/src/main/java/app/andrey_voroshkov/chorus_laptimer/MainActivity.java]: is it granted once and forever?
if it's possible to grant for single app execution, then this place seems better

Comment (SidhNor) [reply Android/ChorusRFLaptimer/app/src/main/java/app/andrey_voroshkov/chorus_laptimer/MainActivity.java]: it's granted until user goes to system settings and revokes the permission. But again, that thing also checks if the permission was already granted, so isn't asking again.

**Pull request merged**